dashboard/legacy/middleware/helpers/profanity_privacy_helper.rb

```diff
@@ -89,6 +89,13 @@ def share_failure_from_body(body, locale)
   blockly_source = parsed_json['source']
   return false unless blockly_source
 
+  # This probably means the filter only works on blockly-based labs
+  # as e.g. Java Lab stores an object in the `main.json` source field like:
+  # blockly_source = {"MyClass.java"=>{"text"=>"my source code for MyClass.java here", "isVisible"=>true, "tabOrder"=>0}}
+  #
+  # See: https://github.com/code-dot-org/code-dot-org/pull/60329#issuecomment-2282270302
+  return false unless blockly_source.is_a? String
+
```

Comment:
our profanity filter was silently not working with non-blockly based labs (object =~ regex wasn't an error, just false), in newer ruby this became an error. After discussion with Sam and Emily, they opted not to change behavior (i.e. not to enable profanity filtering on Java Lab), so that's what this code does.

Reply:
Long discussion on whether to enable this or not here: https://github.com/code-dot-org/code-dot-org/pull/60329#issuecomment-2282270302
And investigation into what broke here: https://github.com/code-dot-org/code-dot-org/pull/60329#issuecomment-2282257696